### PR TITLE
Daily Data Chart: Threshold Lines for Bar Charts, alternate coloring when over threshold

### PR DESCRIPTION
**Pull request by chrisnowak**
## Overview

This adds support for adding an optional set of threshold lines for daily data bar charts.  Colors of bars can now be dynamic based on the value of the bar in relation to the thresholds.

e.g.
Screenshot 2024-03-20 at 11 27 55 AM

This should be useful when setting goals or activity limits to highlight days when activity reached a certain point.

Each threshold allows specifying a threshold value, a reference line color, and and "overThresholdBarColor".  The color of the bar is determined by the highest threshold exceeded by that bar.

Also updates colors in the DailyDataChart to be ColorDefinitions and not strings.

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

N/a there's no unsafe html rendering or anything here

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [ ] [MyDataHelps website](mydatahelps.org)

### Documentation
- [x] I have added relevant Storybook updates to this PR as well.
- [ ] If this feature requires a developer doc update, tag @CareEvolution/api-docs.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

### CODE REVIEW

File: src/components/container/DailyDataChart/DailyDataChart.tsx [hunk 1]
```diff
@@ -241,12 +266,26 @@ export default function DailyDataChart(props: DailyDataChartProps) {
...
+
+                            {(props.options as BarChartOptions)?.thresholds?.map((threshold, index) =>
+                                <linearGradient id={gradientKey + "_threshold" + index} x1="0" y1="0" x2="0" y2="1">
+                                    <stop offset="0%" stopColor={resolveColor(layoutContext.colorScheme, threshold.overThresholdBarColor) || "var(--mdhui-color-warning)"} stopOpacity={1.0} />
+                                    <stop offset="100%" stopColor={resolveColor(layoutContext.colorScheme, threshold.overThresholdBarColor) || "var(--mdhui-color-warning)"} stopOpacity={0.7} />
+                                </linearGradient>
+                            )}
                         </defs>
+                        {(props.options as BarChartOptions)?.thresholds?.filter(t=>t.referenceLineColor)?.map((threshold, index) =>
+                            <ReferenceLine y={threshold.value} stroke={resolveColor(layoutContext.colorScheme, threshold.referenceLineColor)} />
```
Comment (reneefromhold) [inline src/components/container/DailyDataChart/DailyDataChart.tsx:281]: Should a default color be specified for the referenceLine also?

**Review by reneefromhold — APPROVED** [review]
Just one question. lg otherwise

File: src/components/container/DailyDataChart/DailyDataChart.tsx (hunk 1 above)
Comment (chrisnowak) [reply src/components/container/DailyDataChart/DailyDataChart.tsx:281]: If the color isn't specified, the reference line doesn't appear actually

File: src/components/container/DailyDataChart/DailyDataChart.tsx [hunk 2]
```diff
@@ -30,12 +31,19 @@ export interface LineChartOptions {
 }
 
 export interface BarChartOptions {
-    barColor?: string
+    barColor?: ColorDefinition
+    thresholds?: BarChartThreshold[]
+}
+
+export interface BarChartThreshold {
```
Comment (reneefromhold) [inline src/components/container/DailyDataChart/DailyDataChart.tsx:38]: @chrisnowak Something coming to mind, could there be an option where there is a need to highlight under, within, and over range

Comment (reneefromhold) [reply src/components/container/DailyDataChart/DailyDataChart.tsx:38]: I'm mostly thinking about under range. The other 2 (within and over) are already taken care of.

Comment (chrisnowak) [reply src/components/container/DailyDataChart/DailyDataChart.tsx:38]: The default color would apply when under any of the thresholds - so you can totally achieve that with this pretty easily.  E.g. if you want someone to walk between 5000 and 10000 steps per day and highlight days they did it in green, you could set default color to grey, a threshold for 5,000 with a color of green, and then a threshold for 10k with grey again

**Pull request merged**